joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/passes/frontend/XConfigFileCreationPass.scala

```diff
@@ -0,0 +1,75 @@
+package io.joern.x2cpg.passes.frontend
+
+import better.files.File
+import io.shiftleft.codepropertygraph.Cpg
+import io.shiftleft.codepropertygraph.generated.nodes.NewConfigFile
+import io.shiftleft.passes.ConcurrentWriterCpgPass
+import io.shiftleft.semanticcpg.language._
+import io.shiftleft.utils.IOUtils
+import org.slf4j.LoggerFactory
+
+import java.nio.file.Paths
+import scala.util.{Failure, Success, Try}
+
+/** Scans for and inserts configuration files into the CPG.
+  */
+abstract class XConfigFileCreationPass(cpg: Cpg) extends ConcurrentWriterCpgPass[File](cpg) {
+
+  private val logger = LoggerFactory.getLogger(this.getClass)
+
+  // File filters to override by the implementing class
+  protected val configFileFilters: List[File => Boolean]
+
+  private val rootDir = cpg.metaData.root.headOption.getOrElse("<empty>")
+
+  override def generateParts(): Array[File] =
+    if (rootDir.isBlank) {
+      logger.warn("Unable to recover project directory for configuration file pass.")
+      Array.empty
+    } else {
+      Try(File(rootDir)) match {
+        case Success(file) if file.isDirectory =>
+          file.listRecursively
+            .filter(isConfigFile)
+            .toArray
+
+        case Success(file) if isConfigFile(file) =>
+          Array(file)
+
+        case _ => Array.empty
+      }
+    }
+
+  override def runOnPart(diffGraph: DiffGraphBuilder, file: File): Unit = {
+    Try(IOUtils.readLinesInFile(file.path)) match {
+      case Success(fileContentsLines) =>
+        val name       = configFileName(file)
+        val content    = fileContentsLines.mkString("\n")
```

Comment:
Having been bitten by this in the past (https://github.com/joernio/joern/issues/2601#issuecomment-1526054382), viz. files with mixed line break conventions, if you don't really care for lines, there's an alternative API (https://github.com/ShiftLeftSecurity/codepropertygraph/pull/1703) that reads the entire file.

Reply:
Good idea, thanks!